cli/pkg/kube/client.go

```diff
@@ -193,15 +210,14 @@ func (c *Client) ApplyResource(ctx context.Context, configVersion int, obj Objec
 	return err
 }
 
-func (c *Client) DeleteOldOdigosSystemObjects(ctx context.Context, resourceAndNamespace ResourceAndNs, configVersion int) error {
+func (c *Client) DeleteOldOdigosSystemObjects(ctx context.Context, resourceAndNamespace ResourceAndNs, configVersion int, k8sVersion *version.Version) error {
 	systemObject, _ := k8slabels.NewRequirement(odigoslabels.OdigosSystemLabelKey, selection.Equals, []string{odigoslabels.OdigosSystemLabelValue})
 	notLatestVersion, _ := k8slabels.NewRequirement(odigoslabels.OdigosSystemConfigLabelKey, selection.NotEquals, []string{strconv.Itoa(configVersion)})
 	labelSelector := k8slabels.NewSelector().Add(*systemObject).Add(*notLatestVersion).String()
 	resource := resourceAndNamespace.Resource
 	ns := resourceAndNamespace.Namespace
-	k8sVersion, err := odigosver.GetKubernetesVersion()
```

Comment:
I suggest adding a comment to this function, which was the source of the error.
We can mention that it uses `"k8s.io/client-go/rest"` and can only be used for pods that runs inside the cluster.